docs/architecture.md

```diff
@@ -0,0 +1,377 @@
+# Architecture
+
+```mermaid
+flowchart LR
+
+SDK[cow-sdk]
+OrderBookApi
+SubgraphApi
+MetadataApi
+
+SDK --> OrderBookApi
+SDK --> SubgraphApi
+SDK --> MetadataApi
+```
+
+The SDK has 3 main APIs
+
+- **Order Book API**: Allows to get the open orders, historic orders, post new orders, etc.
+- **Subgraph API**: Provides access to On-chain data indexed by The Graph
+- **Metadata API**: Allows to encode/decode meta-data to be attached in orders
+
+## Model: Orders
+
+The orders model used for the API is organized in a hierachy:
+
+```mermaid
+classDiagram
+
+OrderParameters  <|-- OrderCreation
+OrderCreation  <|-- Order
+OrderMetaData  <|-- Order
+Order <|-- EnrichedOrder
+```
+
+## Model: Orders (details)
+
+<style>
+    .important {
+        stroke-width: 3;
+        stoke: #eeff82;
+        fill: #eeff82
+    }
+
+     
+     .important > rect {
+        stroke-width: 3;
+        stoke: #eeff82;
+        fill: #eeff82
+    }
+</style>
```

Comment:
This does not seem to render, at least not in the PR

![image](https://user-images.githubusercontent.com/43217/221587787-715a07f9-b4f6-43f5-8b01-fa3f8d4bed73.png)

Reply:
Removed